src/ray/gcs_rpc_client/accessors/actor_info_accessor.h

```diff
@@ -0,0 +1,244 @@
+// limitations under the License.
+
+#pragma once
+
+#include <functional>
+#include <memory>
+
+#include "absl/synchronization/mutex.h"
+#include "ray/common/gcs_callback_types.h"
+#include "ray/gcs_rpc_client/accessors/actor_info_accessor_interface.h"
+#include "ray/gcs_rpc_client/gcs_client_context.h"
+#include "ray/rpc/rpc_callback_types.h"
+#include "ray/util/sequencer.h"
+
+namespace ray {
+namespace gcs {
+
+using SubscribeOperation = std::function<void(const StatusCallback &done)>;
+using FetchDataOperation = std::function<void(const StatusCallback &done)>;
+
+/**
+  @class ActorInfoAccessor
+
+  Implementation of ActorInfoAccessorInterface that accesses actor information by querying
+  the GCS.
+ */
+class ActorInfoAccessor : public ActorInfoAccessorInterface {
+ public:
+  ActorInfoAccessor() = default;
+  explicit ActorInfoAccessor(GcsClientContext *context);
+  virtual ~ActorInfoAccessor() = default;
+  /**
+    Get actor specification from GCS asynchronously.
+
+    @param actor_id The ID of actor to look up in the GCS.
+    @param callback Callback that will be called after lookup finishes.
+   */
+  void AsyncGet(const ActorID &actor_id,
+                const OptionalItemCallback<rpc::ActorTableData> &callback) override;
+
+  /**
+    Get all actor specification from the GCS asynchronously.
+
+    @param  actor_id To filter actors by actor_id.
+    @param  job_id To filter actors by job_id.
+    @param  actor_state_name To filter actors based on actor state.
+    @param callback Callback that will be called after lookup finishes.
+    @param timeout_ms request timeout, defaults to -1 for infinite timeout.
+   */
+  void AsyncGetAllByFilter(const std::optional<ActorID> &actor_id,
+                           const std::optional<JobID> &job_id,
+                           const std::optional<std::string> &actor_state_name,
+                           const MultiItemCallback<rpc::ActorTableData> &callback,
+                           int64_t timeout_ms = -1) override;
+
+  /**
+    Get actor specification for a named actor from the GCS asynchronously.
+
+    @param name The name of the actor to look up in the GCS.
+    @param ray_namespace The namespace to filter to.
+    @param callback Callback that will be called after lookup finishes.
+    @param timeout_ms RPC timeout in milliseconds. -1 means the default.
+   */
+  void AsyncGetByName(const std::string &name,
+                      const std::string &ray_namespace,
+                      const OptionalItemCallback<rpc::ActorTableData> &callback,
+                      int64_t timeout_ms = -1) override;
+
+  /**
+    Get actor specification for a named actor from the GCS synchronously.
+    The RPC will timeout after the default GCS RPC timeout is exceeded.
+
+    @param name The name of the actor to look up in the GCS.
+    @param ray_namespace The namespace to filter to. NotFound if the name doesn't exist.
+
+    @return Status. TimedOut status if RPC is timed out.
```

Comment:
I prefer doing it like this. I think it's more readable in the code and in the generated docs. 
```suggestion
    @return Status::OK
    @return Status::TimedOut if the method is timed out.
```